flytekit/deck/deck.py

```diff
@@ -174,3 +174,29 @@ def get_deck_template() -> "Template":
         autoescape=select_autoescape(enabled_extensions=("html",)),
     )
     return env.get_template("template.html")
+
+
+class PythonDependencyDeck(Deck):
+    """
+    PythonDependencyDeck is a deck that contains information about the library name and version.
+    """
+
+    def __init__(self, name: Optional[str] = "Python Dependency", html: Optional[str] = ""):
+        super().__init__(name, html)
+
+    @property
+    def html(self) -> str:
+        import json
+        import subprocess
+
+        import pandas as pd
+        from flytekitplugins.deck.renderer import TableRenderer
+
+        try:
+            installed_packages = json.loads(subprocess.check_output(["pip", "list", "--format", "json"]))
```

Comment:
Should that be configurable? not everybody has/uses `pip`

Reply:
I think this PR is already a good first step. If one uses `conda`, `pip` will still pick up all the Python dependencies. The next step is to grab the non-Python `conda` dependencies as well.

I do not think we need to make it configurable. For example, Weights and Bias will automatically pick up all the `pip` dependencies and `conda` dependencies and generate a `requirements.txt` or `environment.yaml`. This allows the user to easily recreate the environment.